lib/charms/data_platform_libs/v0/data_interfaces.py

```diff
@@ -763,14 +763,23 @@ def _process_secret_fields(
         req_secret_fields: Optional[List[str]],
         impacted_rel_fields: List[str],
         operation: Callable,
-        second_chance_as_normal_field: bool = True,
         *args,
         **kwargs,
     ) -> Tuple[Dict[str, str], Set[str]]:
         """Isolate target secret fields of manipulation, and execute requested operation by Secret Group."""
         result = {}
+
+        # If the relation started on a databag, we just stay on the databag
+        # (Rolling upgrades may result in a relation starting on databag, getting secrets enabled on-the-fly)
+        # self.local_app is sufficient to check (ignored if Requires, never has secrets -- works if Provides)
+        fallback_to_databag = (
+            req_secret_fields
+            and self.local_unit.is_leader()
+            and set(req_secret_fields) & set(relation.data[self.local_app])
```

Comment:
can we use `intersection` for better readability?
```suggestion
            and set(req_secret_fields).intersection(set(relation.data[self.local_app]))
```

Reply:
Sure....
(Tho I think it looks a way better as it is ;-) )